src/main/java/org/gridsuite/caseimport/server/CaseImportService.java

```diff
@@ -44,10 +44,11 @@ private String getTargetDirectory(String caseOrigin) {
         return targetDirectory;
     }
 
-    ImportedCase importCaseInDirectory(MultipartFile caseFile, String caseOrigin, String userId) {
+    ImportedCase importCaseInDirectory(MultipartFile caseFile, String caseNameParam, String caseOrigin, String userId) {
         String targetDirectory = getTargetDirectory(caseOrigin);
         UUID caseUuid = caseService.importCase(caseFile);
-        var caseElementAttributes = new ElementAttributes(caseUuid, caseFile.getOriginalFilename(), CASE, new AccessRightsAttributes(false), userId, 0L, null);
+        String caseName = caseNameParam == null || caseNameParam.trim().isEmpty() ? caseFile.getOriginalFilename() : caseNameParam;
```

Comment:
```suggestion
        String caseName = caseNameParam == null || caseNameParam.trim().isEmpty() ? DataSourceUtil.getBaseName(Paths.get(caseFile.getOriginalFilename())) : caseNameParam;
```

Reply:
This would avoid to have this extensions in the name in gridexplore for example. But maybe it rather should be done in case-server